### DISCUSSION TITLE
In android,  Using it as a lib, How to make own app also go proxy traffic？

### DISCUSSION THREAD

**Discussion by sk-huang** [strongswan/strongswan#2417] [Q&A]
I integrated this library into my own application as a lib, this library will exclude my own app by default, resulting in my application traffic is not proxying.

I saw that the code in the CharonVpnService file said "exclude our own app, otherwise the fetcher is blocked", I change the code , added 'break' on the switch (appHandling) of the code, and then the connection failed.  

I want my app to also use proxy traffic, how to set it？

**Comment by tobiasbrunner** [comment]
> and then the connection failed.

Failed how?

**Reply by sk-huang** [reply]
The error message is "Server address lookup failed" .  I checked the history of the code changes,  it's probably the code that ruled out its own app.  
![image](https://github.com/user-attachments/assets/37874804-c8ab-4d9e-b529-b80beaded3eb)
![image](https://github.com/user-attachments/assets/b9600350-dcfe-4a1b-bfba-590bc27dd28f)

What should I do if I want the app that contains this library to also use the vpn？    The strongswan lib version used is latest.

**Reply by tobiasbrunner** [reply]
I guess you could get rid of the blocking TUN device (it's in the same file), but then cleartext traffic will leak while the VPN is getting established.
